bcdaworker/main.go

```diff
@@ -181,36 +181,35 @@ func createDir(path string) error {
 	return nil
 }
 
-func writeBBDataToFile(ctx context.Context, bb client.APIClient, db *gorm.DB, acoID string, acoCMSID string, cclfBeneficiaryIDs []string, jobID, t, since string, transactionTime time.Time) (fileUUID string, error error) {
+func writeBBDataToFile(ctx context.Context, bb client.APIClient, db *gorm.DB, cmsID string, jobArgs models.JobEnqueueArgs) (fileUUID string, err error) {
 	segment := getSegment(ctx, "writeBBDataToFile")
 	defer func() {
 		if err := segment.End(); err != nil {
 			log.Error(err)
 		}
 	}()
 
-	if bb == nil {
-		err := errors.New("Blue Button client is required")
-		log.Error(err)
-		return "", err
-	}
-
-	bbFunc := bbFuncByType(bb, t)
-	if bbFunc == nil {
-		err := fmt.Errorf("Invalid resource type requested: %s", t)
-		log.Error(err)
-		return "", err
-	}
-
-	if !utils.IsUUID(acoID) {
-		err := errors.New("Invalid ACO ID")
-		log.Error(err)
-		return "", err
+	var bundleFunc func(bbID string) (*fhirmodels.Bundle, error)
```

Comment:
This only accepts bbID since all the other `Get...` methods accept arguments in the jobArgs and do not change between beneficiaries.